src/PantheonTerminalWindow.vala

```diff
@@ -676,14 +678,29 @@ namespace PantheonTerminal {
 
             PantheonTerminal.saved_state.tabs = {};
 
+            int focus = 0 <= PantheonTerminal.saved_state.focused_tab < tabs.length
+                            ? PantheonTerminal.saved_state.focused_tab : tabs.length - 1;
             Idle.add_full (GLib.Priority.LOW, () => {
-                foreach (string loc in tabs) {
-                    if (loc == "") {
+                int i;
+                for (i = 0; i < tabs.length; i++) {
+                    if (tabs[i] == "") {
+                        if (i == focus) {
+                            focus++;
+                        }
                         continue;
+                    } else if (focus == i) {
+                        new_tab (tabs[i], null, focus_restored_tabs);
                     } else {
-                        new_tab (loc, null, focus_restored_tabs);
+                        new_tab (tabs[i], null, false);
                     }
                 }
+
+                if (focus_restored_tabs && focus == tabs.length) {
+                    var t = notebook.get_tab_by_index (notebook.n_tabs - 1);
+                    notebook.current = t;
+                    t.grab_focus ();
+                }
```

Comment:
Can the tab be focussed here only? i.e. tabs all created unfocussed and then `var t = notebook.get_tab_by_index (focus.clamp (0, notebook.n_tabs - 1)` etc.